src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests.cs

```diff
@@ -42432,6 +42450,258 @@ static void Test(MyCollection a)
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (11,20): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         object x = null;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(11, 20),
+                // (13,25): warning CS8601: Possible null reference assignment.
+                //         object[] z = [..y];
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "y").WithLocation(13, 25),
+                // (14,16): warning CS8601: Possible null reference assignment.
+                //         z = [..y];
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "y").WithLocation(14, 16));
+        }
+
+        [Fact]
+        public void Spread_Nullable_06()
+        {
+            var source = """
+                #nullable enable
+                using System.Collections.Generic;
+                class Program
+                {
+                    static void Main()
+                    {
+                        IEnumerable<string?> x = [null];
+                        IEnumerable<object> y = [..x];
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (8,36): warning CS8601: Possible null reference assignment.
+                //         IEnumerable<object> y = [..x];
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "x").WithLocation(8, 36));
+        }
+
+        [Fact]
+        public void Spread_Nullable_07()
+        {
+            var source = """
+                #nullable enable
+                using System.Collections.Generic;
+                class Program
+                {
+                    static void Main()
+                    {
+                        IEnumerable<byte> x = [1, 2, 3];
+                        IEnumerable<int?> y = [..x];
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void Spread_Nullable_08()
+        {
+            var source = """
+                #nullable enable
+                using System.Collections.Generic;
+                class Program
+                {
+                    static IEnumerable<T?> F<T>(IEnumerable<T> x)
+                        where T : struct
+                    {
+                        return [..x];
+                    }
+                    static void Main()
+                    {
+                        F<int>([1, 2, 3]);
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+        }
```

Comment:
Do we have a test where the spread operand is more than a simple local? For instance `M(x)`, which could produce nullability warnings of its own.

Reply:
It looks like the operand in test `Spread_Nullable_09` is not producing warnings of its own. It sounded like Julien was asking for an `F()` which warns about a possible null argument, for example.

Reply:
Added case.